packages/integration-tests/src/test-projects/access-testing-project/amplify/backend.ts

```diff
@@ -0,0 +1,120 @@
+import { RemovalPolicy } from 'aws-cdk-lib';
+import {
+  AccountPrincipal,
+  Effect,
+  PolicyDocument,
+  PolicyStatement,
+  Role,
+} from 'aws-cdk-lib/aws-iam';
+
+const backend = defineBackend({
+  auth,
+  data,
+});
+
+const simpleAuthStack = backend.createStack('simple-auth');
+
+const simpleAuthUserPool = new cognito.UserPool(
+  simpleAuthStack,
+  'SimpleAuthUserPool',
+  {
+    removalPolicy: RemovalPolicy.DESTROY,
+  }
+);
+
+const simpleAuthUserPoolClient = new cognito.UserPoolClient(
+  simpleAuthStack,
+  'SimpleAuthUserPoolClient',
+  {
+    userPool: simpleAuthUserPool,
+    authFlows: {
+      userPassword: true,
+    },
+  }
+);
+
+const simpleAuthIdentityPool = new cognito.CfnIdentityPool(
+  simpleAuthStack,
+  'SimpleAuthIdentityPool',
+  {
+    allowUnauthenticatedIdentities: false,
+  }
+);
+
+simpleAuthIdentityPool.cognitoIdentityProviders = [
+  {
+    clientId: simpleAuthUserPoolClient.userPoolClientId,
+    providerName: `cognito-idp.${simpleAuthStack.region}.amazonaws.com/${simpleAuthUserPool.userPoolId}`,
+  },
+];
+
+const iamRolesStack = backend.createStack('iam-roles');
+
+const roleWithAccessToData = new Role(iamRolesStack, 'RoleWithAccessToData', {
+  assumedBy: new AccountPrincipal(iamRolesStack.account),
+  path: '/',
+  inlinePolicies: {
+    root: new PolicyDocument({
+      statements: [
+        new PolicyStatement({
+          actions: [
+            'logs:CreateLogGroup',
+            'logs:CreateLogStream',
+            'logs:PutLogEvents',
+          ],
+          resources: ['arn:aws:logs:*:*:*'],
+          effect: Effect.ALLOW,
+        }),
+      ],
+    }),
+  },
+});
+roleWithAccessToData.applyRemovalPolicy(RemovalPolicy.DESTROY);
+roleWithAccessToData.addToPolicy(
+  new PolicyStatement({
+    actions: ['appsync:GraphQL'],
+    resources: [`${backend.data.resources.graphqlApi.arn}/*`],
+    effect: Effect.ALLOW,
+  })
+);
+
+const roleWithoutAccessToData = new Role(
+  iamRolesStack,
+  'RoleWithoutAccessToData',
+  {
+    assumedBy: new AccountPrincipal(iamRolesStack.account),
+    path: '/',
+    inlinePolicies: {
+      root: new PolicyDocument({
+        statements: [
+          new PolicyStatement({
+            actions: [
+              'logs:CreateLogGroup',
+              'logs:CreateLogStream',
+              'logs:PutLogEvents',
+            ],
+            resources: ['arn:aws:logs:*:*:*'],
+            effect: Effect.ALLOW,
+          }),
+        ],
+      }),
+    },
+  }
+);
+roleWithoutAccessToData.applyRemovalPolicy(RemovalPolicy.DESTROY);
+
+backend.addOutput({
+  custom: {
+    roleWithAccessToDataArn: roleWithAccessToData.roleArn,
+    roleWithoutAccessToDataArn: roleWithoutAccessToData.roleArn,
+    simpleAuthUserPoolId: simpleAuthUserPool.userPoolId,
+    simpleAuthUserPoolClientId: simpleAuthUserPoolClient.userPoolClientId,
+    simpleAuthIdentityPoolId: simpleAuthIdentityPool.ref,
+    authRoleArn: backend.auth.resources.authenticatedUserIamRole.roleArn,
```

Comment:
Aren't the `authRoleArn` and `unauthRoleArn` already exposed as stack outputs? I can't see where these are used.

Reply:
We do not (see [here](https://github.com/aws-amplify/amplify-backend/blob/e2d10a2f93a4bcb477621333f71d133d548034cb/packages/client-config/src/client-config-types/auth_client_config.ts#L5-L34) and [here](https://github.com/aws-amplify/amplify-backend/blob/e2d10a2f93a4bcb477621333f71d133d548034cb/packages/auth-construct/src/construct.ts#L839-L991)) .
that's why I plumbed it via "custom config" to make them visible here https://github.com/aws-amplify/amplify-backend/blob/243a35221be70dd82bbb5fce2ac27a351da59bc8/packages/integration-tests/src/test-project-setup/access_testing_project.ts#L354-L355 and here